### PR TITLE
Upgrade dependencies and node v20

### PR DESCRIPTION
**Pull request by emcfarlane**
Upgrade dependencies and node versions. See the commit-by-commit breakdown for an easier review. First is package json then node updates and finally LICENSE year range with prettier. All code changes are related to formatting and LICENSE years.

Tested here: https://github.com/emcfarlane/test-actions/pull/1

Fixes:
- https://github.com/bufbuild/buf-setup-action/pull/183
- https://github.com/bufbuild/buf-setup-action/pull/173
- https://github.com/bufbuild/buf-setup-action/pull/172
- https://github.com/bufbuild/buf-setup-action/pull/159
- https://github.com/bufbuild/buf-setup-action/pull/149
- https://github.com/bufbuild/buf-setup-action/pull/144
- https://github.com/bufbuild/buf-setup-action/pull/141
- https://github.com/bufbuild/buf-setup-action/pull/140
- https://github.com/bufbuild/buf-setup-action/pull/137

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,22 +10,22 @@
   },
   "engineStrict": true,
   "engines": {
-    "node": ">=16",
+    "node": ">=20",
```
Comment (timostamm) [inline package.json:13]: There is one other reference to Node.js v16 in 
https://github.com/bufbuild/buf-setup-action/blob/71c62afa859a83fef3e99267971d9b67716ff328/.nvmrc

It's used by https://github.com/nvm-sh/nvm and https://github.com/Schniz/fnm

We should bump this to `v20`.

File: package-lock.json [hunk 2]
```diff
@@ -672,9 +828,9 @@
       }
     },
     "node_modules/@typescript-eslint/eslint-plugin/node_modules/semver": {
-      "version": "7.3.7",
-      "resolved": "https://registry.npmjs.org/semver/-/semver-7.3.7.tgz",
-      "integrity": "sha512-QlYTucUYOews+WeEujDoEGziz4K6c47V/Bd+LjSSYcA94p+DmINdf7ncaUinThfvZyu13lN9OY1XDxt8C0Tw0g==",
+      "version": "7.6.0",
```
Comment (timostamm) [inline package-lock.json:831]: On line 2238 in this file, we're still locked to semver v3.6.0.

If you run `npm i semver@6.3.1`, we should be at v3.6.1 or higher everywhere. Can you give this a try?

Comment (emcfarlane) [reply package-lock.json:832]: Thanks for catching, I'll go through and check the rest more thoroughly 😞

Comment (timostamm) [reply package-lock.json:832]: `npm why semver` helps. I only see 6.3.1 and 7.6.0 now, which looks good to me.

**Review by timostamm — APPROVED** [review]
Looks good to me. We update the following dependencies with breaking changes:
- `typescript` and `eslint` - not an issue because the build and lint steps still pass
- `prettier` - new formatting does not affect code
- `@actions/github` - the only breaking change appears to be dropped support for Node 16 (see [changelog](https://github.com/actions/toolkit/blob/main/packages/github/RELEASES.md#600))

**Review by paul-sachs — APPROVED** [review]

**Review by pkwarren — APPROVED** [review]

**Pull request merged**